pkg/schema/schema.go

```diff
@@ -143,10 +158,63 @@ func newCollectionItemValueType(collectionItemValue interface{}) (yamlmeta.Type,
 	return nil, fmt.Errorf("Collection item type did not match any known types")
 }
 
+func DefaultDataValues(doc *yamlmeta.Document) (*yamlmeta.Document, error) {
+	docCopy := doc.DeepCopyAsNode()
+	for _, value := range docCopy.GetValues() {
+		setDefaultValues(value)
+	}
+
+	return docCopy.(*yamlmeta.Document), nil
+}
+
+func setDefaultValues(node interface{}) {
+	switch typedNode := node.(type) {
+	case *yamlmeta.Map:
+		for _, value := range typedNode.Items {
+			setDefaultValues(value)
+		}
+	case *yamlmeta.MapItem:
+		setDefaultValues(typedNode.Value)
+	case *yamlmeta.Array:
+		typedNode.Items = []*yamlmeta.ArrayItem{}
+	}
+}
+
 func (as *AnySchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
 	return yamlmeta.TypeCheck{}
 }
 
+func (n NullSchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
+	return yamlmeta.TypeCheck{}
+}
+
 func (s *DocumentSchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
 	return s.Allowed.AssignTypeTo(typeable)
 }
+
+func (as *AnySchema) AsDataValue() *yamlmeta.Document {
+	return nil
+}
+
+func (n NullSchema) AsDataValue() *yamlmeta.Document {
+	return nil
+}
+
+func (s *DocumentSchema) AsDataValue() *yamlmeta.Document {
+	return s.defaultDVs
+}
+
+func (as *AnySchema) ValidateWithValues(valuesFilesCount int) error {
+	return nil
+}
+
+func (n NullSchema) ValidateWithValues(valuesFilesCount int) error {
+	if valuesFilesCount > 0 {
+		return fmt.Errorf("Schema experiment flag was enabled but no schema document was provided")
```

Comment:
This message feels like it is at the wrong level of abstraction. Should the Schema package be aware that there is a flag in cmd that is experimental?

Reply:
We are going to drop reference to the 'flag' word. And make the error message a bit more generic. "Schema feature is enabled but no schema document was provided"